### DISCUSSION TITLE
Multi Page Form Structure

### DISCUSSION THREAD

**Discussion by shannoga** [pointfreeco/swift-composable-architecture#1937] [Q&A] [answered]
Hello
I need to build a multi page form in which the user answers some questions in each page and submits the for in the last page of the flow. The user can also navigate back to previous pages if he wishes too.
I wish to have a main state that will hold and collect all the input so when submitting I can simply reach for the state and build the request JSON from it. 
But, I wish to avoid handing the full state for each page, each page should scope only the part of the state it needs. At the same time, since each page navigates to the next page it should hand to it the relevant scope that it dose not know anything about.
I am trying to figure out the structure this flow.
Important note - I need to support iOS14+
Would love to get any ideas or example on how you guys would or did achieved that? 
Thanks

**Comment by dannyhertz** [comment, marked as answer]
Until TCA has first class support for this "dynamic" sort of flow (perhaps we might get closer with their new navigation series that is underway!?) I think [TCA Coordinators ](https://github.com/johnpatrickmorgan/TCACoordinators) is a great solution. I have used it to build *many* flows exactly as you describe. Independent screens/features that form a cohesive flow, orchestrated by a parent "flow" feature/reducer. As each screen completes you can gather pieces of state into a larger parent state (or use computed properties to pull state off children as needed to avoid duplicating state in two places). Give it a shot, I think it will suit your needs.

**Reply by shannoga** [reply]
Thanks, will take a look

**Reply by shannoga** [reply]
@dannyhertz 
Just completed building a POC with TCA Coordinators.
Works amazing.
Thanks for your help.
